### PR TITLE
fix submissions to CSV export functionaility

### PR DESCRIPTION
**Pull request by timisenco2015**
# Description
Issues fixed
 - CSV rows backfilling
 - versioning
 - template selection
 - hidden fields not showing in CSV exported 

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have run the npm script lint on the frontend and backend
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have approval from the product owner for the contribution in this pull request

### CODE REVIEW

**Comment by jasonchung1871** [comment]
When exporting, there should be an explanation for what the submission template does. It just says submission template 1, 2, or 3.

**Comment by jasonchung1871** [comment]
Address and tree components do not export

**Comment by timisenco2015** [comment]
> Address and tree components do not export

Yes. I noticed that too. I looking at how Tree component works.

**Comment by timisenco2015** [comment]
> When exporting, there should be an explanation for what the submission template does. It just says submission template 1, 2, or 3.

Can you suggest names that I can use instead? Please.

**Comment by jasonchung1871** [comment]
Address and tree components do not export

> > When exporting, there should be an explanation for what the submission template does. It just says submission template 1, 2, or 3.
> 
> Can you suggest names that I can use instead? Please.

I would use names or have a tooltip explaining what the template does. It's not clear what it does until you export it in each different template and compare them. Even then, I don't know what it's doing exactly.

File: app/frontend/src/store/modules/form.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ const genInitialForm = () => ({
   snake: '',
   submissionReceivedEmails: [],
   userType: IdentityMode.TEAM,
-  versions: [],
+  versions: []
```
Comment (jasonchung1871) [inline app/frontend/src/store/modules/form.js:21]: you can get the form versions from the form itself:

https://github.com/bcgov/common-hosted-form-service/compare/master...jasonchung1871:common-hosted-form-service:export-versioning

**Review by jasonchung1871 — APPROVED** [review]

**Pull request merged**